sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/UnitTestHelper.cs

```diff
@@ -108,9 +108,6 @@ internal static class UnitTestHelper
         public static bool HasExpectedExceptionAttribute(this IMethodSymbol method) =>
             method.GetAttributes().Any(a => a.AttributeClass.IsAny(KnownExpectedExceptionAttributes));
 
-        public static bool IsAssertionMethodName(string name) =>
-            name.SplitCamelCaseToWords().Union(AssertionPartialMethodName).Any();
```

Comment:
This method was doing an `Union` not an `Intersect` and so ALL methods were considered correct.